terraform/helm/aptos-node/files/haproxy.cfg

```diff
@@ -1,47 +1,114 @@
 global
     log stdout len 10240 format raw local0
-    maxconn 500000
-    nbthread 16
+
+    # Config manual: https://cbonte.github.io/haproxy-dconv/2.5/configuration.html
+    # magic values : terraform/helm/aptos-node/values.yaml
+
+    maxconn 1024
+    # This limits the whole HA Proxy impacting both validators and other frontends
+    # maxconnrate 128
+    nbthread 4
+
+    #4MB for client facing sndbuf/rcvbuf. -- 100Mb/s with 300 mili latency (e.g., us-asia)
+    tune.sndbuf.client {{ $.Values.haproxy.limits.validator.tcpBufSize }}
+    tune.rcvbuf.client {{ $.Values.haproxy.limits.validator.tcpBufSize }}
+
     user nobody
 
+## TCP port defaults
 defaults
     log global
-    option tcplog
-    maxconn 500000
-    timeout queue 1s
-    timeout connect 10s
-    timeout server 60s
-    timeout client 60s
-    timeout client-fin 5s
-
-frontend validator
+    mode tcp
+    #option tcplog
+    option dontlog-normal
+    log-format "%ci:%cp - %sp[%rt] [%t] %ft %Tw/%Tc/%Tt %B [%ts] %ac/%fc/%bc/%sc/%rc %sq/%bq"
+    maxconn 1024		#Validator network mesh + FN x2
+    retries 3
+    timeout queue 5s  #limits num of concurrent connections. Not clear if t/o connect is needed. #https://www.papertrail.com/solution/tips/haproxy-logging-how-to-tune-timeouts-for-performance/
+    timeout connect 5s
+    # enough for 1 successfull + 5 unsuccessfull HB(10 sec interval) + 20 sec timeout
+    timeout server 80s
+    timeout client 80s
+
+    timeout client-fin 3s #How long to hold an interrupted client connection.
+    timeout server-fin 1s
+
+frontend fe-{{ include "aptos-validator.fullname" $ }}-validator-RL32
     bind :6180
-    default_backend validator
+    default_backend {{ include "aptos-validator.fullname" $ }}-validator
 
     # Deny requests from blocked IPs
     tcp-request connection reject if { src -n -f /usr/local/etc/haproxy/blocked.ips }
 
-    # Limit to N TCP connections per minute per source IP
-    stick-table type ip size 500k expire 1m store gpc0_rate(1m)
-    tcp-request connection track-sc0 src
-    # TODO: Reject at content phase for now so we get logs, but this should be
-    # done at connection phase for higher efficiency
-    tcp-request content reject if { sc_gpc0_rate(0) ge {{ $.Values.haproxy.limits.validator.connectionsPerIPPerMin }} }
-    tcp-request content sc-inc-gpc0(0) unless { nbsrv(validator) eq 0 }
+    acl ip_high_conn_rate sc0_conn_rate gt {{ $.Values.haproxy.limits.validator.connectionsPerIPPerSec }}
+
+    stick-table type ip size 10m expire 30m store gpt0,gpc0,gpc1,conn_rate(1s),bytes_out_rate(10s),bytes_out_cnt	##about 500MB of memory
```

Comment:
what is gpt, gpc?

Reply:
gpc0 and gpc1 are per connection counters, get is a per connection tag -- currently unusable because of an haproxy bug.